osu.Desktop/Updater/SquirrelUpdateManager.cs

```diff
@@ -183,6 +183,19 @@ private void load(OsuColour colours)
                     }
                 });
             }
+
+            public override void Close()
+            {
+                // cancelling updates is not currently supported by the underlying updater.
+                // only allow dismissing for now.
```

Comment:
I'm so confused by this entire logic. "Cancelling updates is not currently supported" yet you're switching on the cancelled state? How does it ever get into the cancelled state?

Reply:
It can happen if the updater fails, search for usages.

Reply:
Yikes that's some convoluted logic, and all of this being in the `Close()` method... Like I can understand how it works now but I had to go through 4 classes to figure out why everything's the way it is.

Also, I take it progress notifications never get into the completed state, which is why the switch doesn't consider it?

Reply:
Progress notifications immediately dismiss on completion, showing their completion notification, yep.